tests/PSRule.Rules.Azure.Tests/Azure.AppGwWAF.Tests.ps1

```diff
@@ -88,14 +88,14 @@ Describe 'Azure.AppGWWAF' -Tag 'Network', 'AppGwWAF' {
             # Fail
             $ruleResult = @($filteredResult | Where-Object { $_.Outcome -eq 'Fail' });
             $ruleResult | Should -Not -BeNullOrEmpty;
-            $ruleResult.Length | Should -Be 1;
-            $ruleResult.TargetName | Should -Be 'appgwwaf-C';
+            $ruleResult.Length | Should -Be 2;
+            $ruleResult.TargetName | Should -Be 'appgwwaf-A', 'appgwwaf-B';
 
             # Pass
             $ruleResult = @($filteredResult | Where-Object { $_.Outcome -eq 'Pass' });
             $ruleResult | Should -Not -BeNullOrEmpty;
-            $ruleResult.Length | Should -Be 2;
-            $ruleResult.TargetName | Should -Be 'appgwwaf-A', 'appgwwaf-B';
+            $ruleResult.Length | Should -Be 1;
+            $ruleResult.TargetName | Should -Be 'appgwwaf-C';
```

Comment:
For more complete testing let's test each condition separately by making the resource data:

- `appgwwaf-A` with OWASP `3.2`, Bot Manager `0.1`, which should fail.
- `appgwwaf-B` with OWASP `3.1`, Bot Manager `1.0`, which should fail.
- `appgwwaf-C` with OWASP `3.2`, Bot Manager `1.0`, which would pass.